desktop/core/src/desktop/js/reactComponents/FileChooser/PathBrowser/PathBrowser.tsx

```diff
@@ -0,0 +1,113 @@
+// Licensed to Cloudera, Inc. under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  Cloudera, Inc. licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React, { useRef, useEffect, useState, RefObject } from 'react';
+import { Breadcrumb, Input } from 'antd';
+
+import HdfsIcon from '../../../components/icons/HdfsIcon';
+import S3Icon from '../../../components/icons/S3Icon';
+import AdlsIcon from '../../../components/icons/AdlsIcon';
+
+import { BreadcrumbData } from '../types';
+import './PathBrowser.scss';
+
+interface PathBrowserProps {
+  breadcrumbs?: BreadcrumbData[];
+  onFilePathChange: (path: string) => void;
+}
+
+const PathBrowser: React.FC<PathBrowserProps> = ({ breadcrumbs, onFilePathChange }) => {
+  const [inputFieldDisplay, setInputFieldDisplay] = useState(false);
+
+  const icons = {
+    //hdfs file system begins with the first breadcrumb as "" (ex: /user/demo)
+    '': <HdfsIcon />,
+    abfs: <AdlsIcon />,
+    s3: <S3Icon />
+  };
+
+  const useOutsideAlerter = (ref: RefObject<HTMLDivElement>) => {
+    useEffect(() => {
+      // Alert if clicked on outside of element
+      const handleClickOutside = (event: MouseEvent) => {
+        const current = ref?.current;
+        if (current && !current.contains(event.target as Node)) {
+          setInputFieldDisplay(false);
+        }
+      };
+      // Bind the event listener
+      document.addEventListener('mousedown', handleClickOutside);
+      return () => {
+        // Unbind the event listener on clean up
+        document.removeEventListener('mousedown', handleClickOutside);
+      };
+    }, [ref]);
```

Comment:
Have you verified that this works? ref changes normally don't trigger a rerender....

Reply:
Yes it works as I showed during todays meet.

Reply:
The code executes, but I don't think your dependency array (the second argument when you create a react hook) does what it might look like here. The dependency array is suppose to inform the hook for which changes it needs to rerun, but I think this hook will only run once since changes to refs won't trigger a component rerender. Adding `[ref]` is the same as adding an empty dependency array, which by the way is okay in this scenario since the containing element that we are referencing won't change during the lifetime of the component instance.

It is perhaps better explained here :-) https://epicreact.dev/why-you-shouldnt-put-refs-in-a-dependency-array/